web/html-forms.md

```diff
@@ -0,0 +1,164 @@
+
+```
+<form>
+    <select id="fruits" name="fruits">
+    <option value="apple">Apple</option>
+    <option value="mango">Mango</option>
+    <option value="banana">Banana</option>
+    </select>
+</form>
+```
+
+3. Textarea Element : It can be seen as an input field only which allows text in multiple lines.
+
+Following is an example which shows the code snippet to declare the Textarea Element.
+
+```
+<form>
+    <label for="message">Message:</label>
+    <textarea rows="2" cols="30" name="message"  id="message"></textarea>
+</form>
+```
+
+4. Button Element : It is a clickable element which is usually used to submit the data.
+
+Following is an example which shows the code snippet to declare the Button Element.
+
+```
+<form>
+    <button type="button">Click Me!</button>
+</form>
+```
+
+#### Validation Attribute
+HTML elements which allows restriction in the form elements are the validation attribute.
+1. Readonly Attribute : This attribute do not allow any text to be entered in the input field.
+```
+<form>
+  <label for="name">Name:</label><br>
+  <input type="text" id="name" name="name" value="Shreya" readonly><br>
+  <label for="email">Email:</label><br>
+  <input type="email" id="email" name="email" value="shreyakapoor98@gmail.com">
+</form>
+```
+
+2. Disabled Attribute : In this the input field is disabled and we cannot click on it. 
+```
+<form>
+  <label for="name">Name:</label><br>
+  <input type="text" id="name" name="name" value="Shreya" disabled><br>
+  <label for="email">Email:</label><br>
+  <input type="email" id="email" name="email" value="shreyakapoor98@gmail.com">
+</form>
+```
+
+3. Size Attribute: This attribute defines the width of the input field.
+```
+<form>
+  <label for="name">Name:</label><br>
+  <input type="text" id="name" name="name" size="40"><br>
+  <label for="number">Number:</label><br>
+  <input type="number" id="number" name="number" size="4">
+</form>
+```
+
+#### Submit and Action for Non AJAX Form
+After the user enters the details, the data or the information is passed for further processing. 
+Input field of type submit is taken to submit the data and action is specified corresponding to it. In case of non AJAX forms the action is GET.
+The get request includes all the data in the URL.
+Following code snippet shows the illustration of submission of form.
+```
+<form action="/submit_action.php" method="get">
+  <label for="name">Name:</label>
+  <input type="text" id="name" name="name">
+  <label for="address">Address:</label>
+  <input type="text" id="address" name="address">
+   <input type="submit" value="Submit">
+</form>
+```
+
+### Common Mistakes / Misconceptions
+
+- **Javascript must be known to make an HTML form**
+Only basic knowledge of HTML is required to create a form in HTML.
+
+### Independent Practice
+Try making out forms and apply the concepts which you have learnt from this. [Click here to practice](https://codepen.io/lauralikespi/pen/WJaqJe) .
```

Comment:
Please add some other link for independent practice which displays only a basic form with minimal css and javascript.